includes/Products.php

```diff
@@ -497,8 +497,107 @@ public static function update_commerce_enabled_for_product( \WC_Product $product
 	 */
 	public static function get_google_product_category_id( \WC_Product $product ) {
 
-		// TODO: implement
-		return '';
+		// attempt to get from product or parent product metadata
+		if ( $product->is_type( 'variation' ) ) {
+			$google_product_category_id = get_post_meta( $product->get_parent_id(), self::GOOGLE_PRODUCT_CATEGORY_META_KEY, true );
+		} else {
+			$google_product_category_id = $product->get_meta( self::GOOGLE_PRODUCT_CATEGORY_META_KEY );
+		}
+
+		// fallback to the highest category's Google product category ID
+		if ( empty( $google_product_category_id ) ) {
+
+			$google_product_category_id = self::get_google_product_category_id_from_highest_category( $product );
+		}
+
+		// fallback to plugin-level default Google product category ID
+		if ( empty( $google_product_category_id ) ) {
+
+			$google_product_category_id = facebook_for_woocommerce()->get_commerce_handler()->get_default_google_product_category_id();
+		}
+
+		return $google_product_category_id;
+	}
+
+
+	/**
+	 * Gets the stored Google product category ID from the highest category.
+	 *
+	 * @since 2.1.0-dev.1
+	 *
+	 * @param \WC_Product $product the product object
+	 * @return string
+	 */
+	private static function get_google_product_category_id_from_highest_category( \WC_Product $product ) {
+
+		$google_product_category_id = '';
+
+		// get all categories for the product
+		$categories = get_the_terms( $product->get_id(), 'product_cat' );
```

Comment:
this won't get the terms if it's a variable product will it? I think you'll need to either get the parent in here or before calling this method.